src/rougelikeLibrary/Position.java

```diff
@@ -0,0 +1,57 @@
+package rougelikeLibrary;
+
+import java.util.Objects;
+
+public class Position {
+	private int x, y;
+
+	public Position(int x, int y) {
+		if (x < 0 || y < 0) {
+			throw new IllegalArgumentException("X or Y can't be negative.");
+		}
+
+		this.x = x;
+		this.y = y;
+	}
+
+	@Override
+	public String toString() {
+		return "X: " + x + " Y: " + y;
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(x, y);
```

Comment:
Egentligen ska man sätta in sin egen Hash funktion, annars är det 100% meningslöst att override metoden.

Reply:
hash
public static int hash(Object... values)
Generates a hash code for a sequence of input values. The hash code is generated as if all the input values were placed into an array, and that array were hashed by calling Arrays.hashCode(Object[]).
This method is useful for implementing Object.hashCode() on objects containing multiple fields. For example, if an object that has three fields, x, y, and z, one could write:

 @Override public int hashCode() {
     return Objects.hash(x, y, z);
 }
Från http://docs.oracle.com/javase/7/docs/api/java/util/Objects.html#hash%28java.lang.Object...%29

Så jag implementerar en ny hash som fungerar med equal.